specs/grafana_event_handling_in_monitoring_integration.adoc

```diff
@@ -0,0 +1,327 @@
+:imagesdir: ./images
+
+= Grafana Event Handling In Monitoring-Integration
+
+Alerting system is required to set alert-conditions based on user-defined thresholds
+and notify alerts to the corresponding user. As per the new architecture the thresholds 
+for alerts will be configured in grafana and all events of alert will be notified to
+respective user via the tendrl.
+
+
+== Problem description
+
+In the present alerting scenerio ,the alerts are raised by the tendrl component which monitors the system.
+Tendrl alerting module notifies respective users regarding these alerts.The notifications send by this module
+contains properly structured alert details including some tendrl specific metadata. In the new architecture,
+grafana has been introduced to monitor the system and henceforth all alerts will be configured in grafana,
+but the alert structure received from grafana lacks important alerting details and tendrl specific
+details required by the user.
+
+== Use cases
+
+Goal: Notify user in case of any alert event.
+
+Actors:
+
+1) User 
+
+Steps:
+
+* An alert is configured in grafana as per user requirement.
+
+* A Notification channel is configured in grafana.
+
+* Any change in system state is caught by tendrl component
+  via alerts sent by grafana.
+
+* The alert structure received from grafana is converted into current alert 
+  structure and passed to node agent socket.
+
+* The new alerting event is sent to respective user via tendrl component called notifier.
+
+
+== Proposed change
+
+
+image::alert_component.png[Tendrl Alerting System]
+
+
+Alert-Handler: It will receive alerts from Grafana and modify the alerts as per tendrl specifications.
+Notifier: Its function is to send notification to the users.(The present alert system)
```

Comment:
Link to the spec instead of the text in the parentheses (https://github.com/Tendrl/specifications/issues/190).